src/openloco/ui/WindowManager.cpp

```diff
@@ -541,36 +579,294 @@ namespace openloco::ui::WindowManager
         return (window*)regs.esi;
     }
 
-    // 0x004C9C68
+    /**
+     * 0x004C9BEA
+     *
+     * @param x @<dx>
+     * @param y @<ax>
+     * @param width @<bx>
+     * @param height @<cx>
+     */
+    static bool window_fits_within_space(gfx::point_t position, gfx::ui_size_t size)
+    {
+        if (position.x < 0)
+            return false;
+
+        if (position.y < 28)
+            return false;
+
+        if (position.x + size.width > ui::width())
+            return false;
+
+        if (position.y + size.height > ui::width())
+            return false;
+
+        for (ui::window* w = &_windows[0]; w != _windowsEnd; w++)
+        {
+            if ((w->flags & window_flags::stick_to_back) != 0)
+                continue;
+            if (position.x + size.width <= w->x)
+                continue;
+            if (position.x > w->x + w->width)
+                continue;
+            if (position.y + size.height <= w->y)
+                continue;
+            if (position.y >= w->y + w->height)
+                continue;
+
+            return false;
+        }
+
+        return true;
+    }
+
+    // 0x004C9F27
+    static window* foundSpace(
```

Comment:
`foundSpace` sounds a bit vague to me. I realise it's temporary until we can use a std container, but I think `emptyWindowSlot` or something might be clearer?